### ISSUE TITLE
Sporadic test failures with 1.15

### ISSUE THREAD

**Issue by hhoffstaette** [miquels/liblockfile#3]
In Gentoo we noticed that the new test suite seems to fail quite often - probably because
it's tied to the build process by default, getting exercised more often.
The bug is [here](https://bugs.gentoo.org/691706).

After improving the test failure message to include the test number, the following can be
observed after a default ./configure && make liblockfile.so:

```
$for i ($(seq 100)) ./run-tests.sh
tests OK
tests OK
2) lockfile still exists after unlock
tests OK
tests OK
tests OK
2) lockfile still exists after unlock
tests OK
2) lockfile still exists after unlock
tests OK
tests OK
tests OK
tests OK
2) lockfile still exists after unlock
tests OK
tests OK
2) lockfile still exists after unlock
tests OK
tests OK
2) lockfile still exists after unlock
tests OK
tests OK
tests OK
2) lockfile still exists after unlock
2) lockfile still exists after unlock
tests OK
2) lockfile still exists after unlock
tests OK
2) lockfile still exists after unlock
tests OK
tests OK
tests OK
tests OK
tests OK
2) lockfile still exists after unlock
tests OK
tests OK
tests OK
tests OK
2) lockfile still exists after unlock
tests OK
tests OK
2) lockfile still exists after unlock
tests OK
tests OK
2) lockfile still exists after unlock
tests OK
tests OK
tests OK
tests OK
2) lockfile still exists after unlock
tests OK
2) lockfile still exists after unlock
tests OK
tests OK
tests OK
tests OK
tests OK
2) lockfile still exists after unlock
tests OK
tests OK
<..etc..>
```
So it's always the second test that fails. I can reproduce this regardless of compiler,
CFLAGS, filesystem or shell.

**Comment by valebes** [comment]
I have the same problem on gentoo

**Comment by miquels** [comment]
Yes, the debian autobuilders also ran into that problem. For now, I'm not running the tests automatically after every build anymore until I can write better tests :) "master" has these changes (and a lot more).

**Comment by yasuhirokimura** [comment]
I'm a maintainer of FreeBSD liblockfile port.
On FreeBSD `run-test.sh` of 1.15 always fails with same error.
And `run-test.sh` of master fails with "double lock while running command".

**Comment by miquels** [comment]
I released 1.16. should fix things.

**Comment by hhoffstaette** [comment]
Ran the tests again in a a loop and now without a single failure - thanks!
